tests/phpunit/api/v3/ACLPermissionTest.php

```diff
@@ -95,7 +95,21 @@ public function testContactGetNoResultsHookWithEditMyContact() {
       'check_permissions' => 1,
       'return' => 'display_name',
     ));
-    $this->assertEquals(0, $result['count']);
+    $this->assertEquals(1, $result['count']);
+  }
+
+  /**
+   * Function tests that a user with "edit my contact" can edit themselves.
+   */
+  public function testContactEditHookWithEditMyContact() {
+    $this->markTestIncomplete('api acls only work with contact get so far');
+    $cid = $this->createLoggedInUser();
+    $this->hookClass->setHook('civicrm_aclWhereClause', array($this, 'aclWhereHookNoResults'));
+    CRM_Core_Config::singleton()->userPermissionClass->permissions = array('access CiviCRM', 'edit my contact');
+    $this->callAPISuccess('contact', 'create', array(
+      'check_permissions' => 1,
+      'id' => $cid,
+    ));
```

Comment:
My guess is that we would remove the permission on add contact & then in the contact_create function we would call a function (which I'm sure exists) that confirms the contact has permission to edit that contact if ID is set.

For create I presume it would fall back to whatever blanket permission it is currently using.

I expect this will become an important change soon with the increasing front-end use of the api